thrust/thrust/system/omp/detail/scan.h

```diff
@@ -106,7 +106,9 @@ OutputIterator scan_impl(
 
     if (start < n)
     {
-      block_sums[tid] = ::cuda::std::reduce(first + start, first + end, accum_t{}, wrapped_binary_op);
+      // For both has_init and no-init cases: reduce each block using first element as init
+      accum_t first_elem = *(first + start);
+      block_sums[tid]    = ::cuda::std::reduce(first + start + 1, first + end, first_elem, wrapped_binary_op);
```

Comment:
Important: Let's avoid an extra copy:
```suggestion
      block_sums[tid]    = ::cuda::std::reduce(first + start + 1, first + end, *(first + start), wrapped_binary_op);
```